src/stream/src/executor/source/source_executor.rs

```diff
@@ -248,6 +243,16 @@ impl<S: StateStore> SourceExecutor<S> {
             .await
             .unwrap();
 
+        let source_desc = self
+            .source_builder
+            .build()
+            .await
+            .context(anyhow!("build source desc failed"))?;
+        self.row_id_index = source_desc
```

Comment:
Better to add some comments on why we need this transformation. :)

Reply:
hhh